tests/test_cmab_meta_model.py

```
@@ -0,0 +1,244 @@
+) -> None:
+    """An arm absent from the update batch keeps its head posterior unchanged."""
+    mu_before = _head_mu(meta_no_backbone, ABSENT_ARM).copy()
+    actions: List[ActionId] = sorted(VALID_SUBSET) * (N_ROWS // 2)
+    rewards = rng.integers(0, 2, size=len(actions)).tolist()
+    meta_no_backbone.update(actions=actions, rewards=rewards, context=_context(rng, n_rows=len(actions)))
+    np.testing.assert_array_equal(_head_mu(meta_no_backbone, ABSENT_ARM), mu_before)
+
+
+def test_meta_model_requires_at_least_one_action() -> None:
+    """An empty actions dict is rejected at construction."""
+    with pytest.raises((ValueError, AttributeError), match="[Aa]t least one action"):
```

Ruff (0.15.17)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)